### PR TITLE
Created a way to aquire a plex token via sign in instead

### PR DESCRIPTION
**Pull request by notbaab**
Adds a way to create a plex instance with a plex.tv username and password by requesting a token according to [this](https://forums.plex.tv/discussion/129922/how-to-request-a-x-plex-token-token-for-your-app/p1) specification.

Technically, the client identification needs to be a unique identifier though I'm not sure how best to do that. 

I don't write much go code so be gentle...

### CODE REVIEW

File: models.go [hunk 1]
```diff
@@ -709,6 +709,33 @@ type BaseAPIResponse struct {
...
+		Plan     string   `json:"lifetime"`
+		Features []string `json:"features"`
+	} `json:"subscription"`
+	Roles struct {
+		Roles []string `json:"roles"`
+	} `json:"roles"`
+	Entitlements []string `json:"entitlements"`
+	ConfirmedAt  string   `json:"confirmedAt"`
+	ForumID      int      `json:"forumId"`
+	RemeberMe    bool     `json:"rememberMe"`
```
Comment (jrudio) [inline models.go]: Should be `RememberMe`

:)

File: plex.go [hunk 2]
```diff
@@ -49,6 +50,55 @@ func New(baseURL, token string) (*Plex, error) {
...
 }
 
+// Creates a plex instance using a user name and password instead of an auth
+// token.
+func SignIn(baseURL, username, password string) (*Plex, error) {
+	if baseURL == "" {
+		return &Plex{}, errors.New("url is required")
+	}
+
+	_, err := url.ParseRequestURI(baseURL)
```
Comment (jrudio) [inline plex.go]: We should check the error here

```go
  if _, err := url.ParseRequestURI(baseURL); err != nil {
      return &Plex{}, err
  }
```

File: models.go (hunk 1 above)
Comment (notbaab) [inline models.go]: Oops. It's literally right next to it spelled correctly.

**Comment by jrudio** [comment]
Thanks a lot, man!

**Pull request merged**